From 0e45560ab829a28b28e16c127f38ee1315333c4c Mon Sep 17 00:00:00 2001
From: Hannu Mallat <hannu.mallat@jollamobile.com>
Date: Fri, 29 Aug 2014 10:00:34 +0300
Subject: [PATCH 3/9] ipconfig: Do not pass a NULL pointer to D-Bus

connman_inet_ifname() may return a NULL pointer e.g. if the interface
does not exist in the kernel anymore but connman isn't yet aware of
that. Passing a NULL pointer to D-Bus may result in an abort, so check
the return value.

(cherry picked from commit 221fcf32c38d8b6b7f44f8a00e22167e4d1e0e98)
Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 src/ipconfig.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/ipconfig.c b/src/ipconfig.c
index b7e4f94..fa4c0d6 100644
--- a/src/ipconfig.c
+++ b/src/ipconfig.c
@@ -2166,9 +2166,11 @@ void __connman_ipconfig_append_ethernet(struct connman_ipconfig *ipconfig,
 
 	if (ipconfig->index >= 0) {
 		char *ifname = connman_inet_ifname(ipconfig->index);
-		connman_dbus_dict_append_basic(iter, "Interface",
-					DBUS_TYPE_STRING, &ifname);
-		g_free(ifname);
+		if (ifname) {
+			connman_dbus_dict_append_basic(iter, "Interface",
+						DBUS_TYPE_STRING, &ifname);
+			g_free(ifname);
+		}
 	}
 
 	if (ipdevice->address)
-- 
1.9.1

